### PR TITLE
[CCXDEV-11073] OpenAPI specification for on-demand data gathering endpoints

### PR DESCRIPTION
**Pull request by epapbak**
# Description

- Add specification for the new endpoints in v2/openapi.json

Fixes CCXDEV-11073

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing steps

CI

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: server/api/v2/openapi.json [hunk 1]
```diff
@@ -1744,6 +1988,32 @@
           }
         }
       },
+      "simplifiedReport": {
+        "type": "array",
+        "items": {
+          "type": "object",
+          "properties": {
+            "rule_fqdn": {
```
Comment (Bee-lee) [inline server/api/v2/openapi.json:1996]: is rule_fqdn an agreed name? IIRC It's not used anywhere in our API and people always have different understanding of this name. Most people expect the error key too if it's called "fully qualified......"

File: server/api/v2/openapi.json [hunk 2]
```diff
@@ -357,6 +346,254 @@
...
+            "description": "Invalid request (e.g cluster ID with unexpected format)"
+          },
+          "404": {
+            "description": "Cluster not found"
+          }
+        }
+      },
+      "post": {
+        "summary": "Filtered list of stored requests for given cluster ID",
+        "description": "A list of requests for given cluster, filtered by the provided request IDs, if any.",
```
[the remark this answers is not in the document]
Comment (epapbak) [reply server/api/v2/openapi.json:402]: With the specification as is, it should return a `400 Bad request` if the POST endpoint is used without providing at least one request ID.

File: server/api/v2/openapi.json (hunk 1 above)
Comment (epapbak) [reply server/api/v2/openapi.json:1996]: Let me run that through the right people and come back to you

Comment (epapbak) [reply server/api/v2/openapi.json:1996]: We have some inconsistencies when it comes to naming rules (with or without error key, with '.report', etc) due supporting different clients, but we kind of agreed on using `rule_fqdn|erorr_key` combination and call it `rule_selector` (e.g v2 `clusters_detail` endpoint).

The `rule_fqdn` name comes from the [rules specification](https://docs.google.com/document/d/1bGwR1J7bezb2k-KbJ5lVtEd4ftq_9WSVK6vd_3MllwY/edit#heading=h.i00d53mlijgv) itself.

**Review by Bee-lee — APPROVED** [review]

**Review by tisnik — APPROVED** [review]
ok

**Pull request merged**